### PR TITLE
Add support for picking up proxy info from standard env vars.

### PR DESCRIPTION
**Pull request by sundbp**
This change aims to make deps.clj pick up proxy info from env variables and pass it on to the "standard" JVM options. This enables dpes.clj to work behind an (unauthenticated) proxy to pull down dependencies etc (something the bash shell script does not do currently).

### CODE REVIEW

**Comment by borkdude** [comment]
@sundbp I just created a protection rule for the master branch. Can you add a fake / empty commit to trigger it? I will squash the branch anyway, so there is no need for force-pushing. 

Is it possible to add some sort of unit test for this? I assume you tested this yourself? Should we add docs for this?

**Comment by sundbp** [comment]
Ok, will push a commit adding a a comment to the README of this. May be tonight, about to head out a few hours.

I'll think of a unit test. Not super obvious to me but perhaps there's something to be done.

**Comment by borkdude** [comment]
> I'll think of a unit test. Not super obvious to me but perhaps there's something to be done.

Maybe just a test that runs the `jvm-proxy-settings` function using `clojure.java.shell/with-sh-env` and returns the right output?

File: deps.clj [hunk 1]
```diff
@@ -216,6 +216,47 @@ function Get-StringHash($str) {
     (unzip dest (.getPath deps-clj-config-dir))
     (.delete dest)))
 
+(def ^:private authenticated-proxy-re #".+:.+@(.+):(\d+).*")
```
Comment (borkdude) [inline deps.clj]: Why dynamic vars?

File: src/borkdude/deps.clj [hunk 2]
```diff
@@ -216,6 +216,47 @@ function Get-StringHash($str) {
     (unzip dest (.getPath deps-clj-config-dir))
     (.delete dest)))
 
+(def ^:private authenticated-proxy-re #".+:.+@(.+):(\d+).*")
```
Comment (borkdude) [inline src/borkdude/deps.clj]: Why dynamic vars?

Comment (sundbp) [reply src/borkdude/deps.clj]: Not dynamic, just private. I probably had them used in more than 1 place originally and hence had them in a constant def. The final version only has them in 1 place so can remove the def's and put them where they're used. Got spot.

**Comment by sundbp** [comment]
> > I'll think of a unit test. Not super obvious to me but perhaps there's something to be done.
> 
> Maybe just a test that runs the `jvm-proxy-settings` function using `clojure.java.shell/with-sh-env` and returns the right output?

Yeah - that's reasonable. My proxy url parsing is hardly extensive, but I have never seen anything that isn't just http://host:port in reality (or the authenticated version of the same). So I'm just supporting that - let me know if you know of any variations or "standard" schema for proxy info.

File: src/borkdude/deps.clj (hunk 2 above)
Comment (borkdude) [reply src/borkdude/deps.clj]: Sorry, misread. This is fine.

**Comment by borkdude** [comment]
@sundbp I added a TODO commit here:

https://github.com/borkdude/deps.clj/pull/8/commits/91cf44f7c7cec18b75930eddd2325500695b17d7

If you could complete that unit test and write something in the TODO spot in the README, I'll merge this and make a release. I was going to make a new release anyway since `clojure` itself made some fixes today.

**Comment by borkdude** [comment]
@sundbp I just merged it so I can release. Can you add the test + docs in a separate PR?

**Comment by sundbp** [comment]
Thanks. OK, I'll PR something in the morning.

**Comment by sundbp** [comment]
Btw, I haven't forgotten about this - I'll get to it!

**Comment by borkdude** [comment]
@sundbp I hope so! :)

**Pull request merged**